### DISCUSSION TITLE
Loop statements could be with a loop variable

### DISCUSSION THREAD

**Discussion by xp44mm** [fsharp/fslang-suggestions#1243] [General]
If we redesign the f # language in the future, the loop statement should contain a loop variable, which is an integer, starting from 0, with a step size of 1.  The semantics of the loop variable is to indicate the times of loops.

Only two loops are reserved, first is `for in`, another is `while`.

`for in` It uses similar to the curried interface of the `iteri` loop:

```f#
for i e in ['a'..'d'] do
    Console.Write(sprintf "%d" i)
//01234
```

loop `while`:

```f#
for i while i<5 do
    Console.Write(sprintf "%d" i)
//01234
```

An infinite loop with a loop variable, which is implemented by omitting the `while` clause of the while loop.

```f#
for i do
    dosomething i
```

They can still simplify the compatibility of corresponding statements without loop variables.

```f#
for e in ['a'..'d'] do
    Console.Write(sprintf "%A" e)
```

loop `while`:

```f#
while truly() do
    Console.Write("do something.")
```

**Comment by Xyncgas** [comment]
System.Linq or F# stanard library provides this function
it's shorter, ```mapi``` ```iteri``` etc or write your own function

**Reply by Xyncgas** [reply]
```F#
for i do
    dosomething i
```
Given i wasn't given a definition, An automatic increment of it, Yes it's too simple to be happening in F#
